### PR TITLE
Fix to_account calculation when payment processor is present.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Possible regression - but let's not hold 5.18.4 over it - targetting 5.19.0 seems fine since more digging still needed & I just spotted it trying to document stuff

I noticed this but haven't tested. However, it seems like this would potentially get the wrong account sometimes.

Fix properly for master

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -955,8 +955,9 @@ public static function recordPaymentActivity($contributionId, $participantId, $t
    * @return int
    */
   public static function getToFinancialAccount($contribution, $params) {
-    if (!empty($params['payment_processor'])) {
-      return CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contribution['payment_processor'], NULL, 'civicrm_payment_processor');
+    $paymentProcessorID = $params['payment_processor_id'] ?? CRM_Utils_Array::value('payment_processor', $params);
+    if (!empty($paymentProcessorID)) {
+      return CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($paymentProcessorID, NULL, 'civicrm_payment_processor');
```
Comment (mattwire) [inline CRM/Contribute/BAO/Contribution.php]: @eileenmcnaughton There is a change here.  Previously this line was getting the payment processor ID from the `$contribution` array and now it's getting it from the `$params` array.

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: Yeah - I think it's more correct but I'm going to write a test & do a bit more work on this.  I'm targetting the 5.19 rc but am not going to try to do a 5.18 drop on this - I think that's the right level of 'rush'

**Comment by eileenmcnaughton** [comment]
I'm Ok to close this & do it against master - it perhaps should go on 5.19 but we've got a lot of related cleanup already in 5.20 - but I think this PR https://github.com/civicrm/civicrm-core/pull/15640 does need to be merged before the rc gets cut

**Comment by eileenmcnaughton** [comment]
Addressed in 5.20 as it's kinda complex & it seemed OK not to rush into the rc

**Pull request closed without merging**